addons/sale/models/sale_order_line.py

```diff
@@ -291,7 +291,17 @@ def _compute_name(self):
         for line in self:
             if not line.product_id:
                 continue
-            line.name = line.with_context(lang=line.order_partner_id.lang)._get_sale_order_line_multiline_description_sale()
+
+            name = line.with_context(lang=line.order_partner_id.lang)._get_sale_order_line_multiline_description_sale()
+            if line.is_downpayment and not line.display_type:
+                dp_state = line._get_downpayment_state()
+                if dp_state == 'draft':
+                    name = _("%(line_description)s (Draft)", line_description=name)
+                elif dp_state == 'cancel':
+                    name = _("%(line_description)s (Canceled)", line_description=name)
+                elif dp_state == 'posted':
+                    name = _("%(line_description)s (Posted)", line_description=name)
```

Comment:
```suggestion
                context = {'lang': line.order_partner_id.lang}
                dp_state = line._get_downpayment_state()
                if dp_state == 'draft':
                    name = _("%(line_description)s (Draft)", line_description=name)
                elif dp_state == 'cancel':
                    name = _("%(line_description)s (Canceled)", line_description=name)
                elif dp_state == 'posted':
                    name = _("%(line_description)s (Posted)", line_description=name)
                del context
```
You need a local context variable with the lang to make sure the translation is done in the customer language.